client/src/components/select-formula-view/SelectFormulaForm.js

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import FormControl from '@mui/material/FormControl';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import FormLabel from '@mui/material/FormLabel';
+import InputLabel from '@mui/material/InputLabel';
+import Select from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import Typography from '@mui/material/Typography';
+import styled from 'styled-components';
+import ViewFormulaAccordion from './ViewFormulaAccordion';
+
+const StyledBox = styled(Box)`
+  width: 53vw;
+  min-width:  400px;
+  max-width: 1000px;
+`;
+
+const SelectFormulaForm = ({ assignments, formulas, instaceId }) => {
+
+  const [formula, setFormula] = useState('');
+  const [codeSnippet, setCodeSnippet] = useState('');
+  const [selectedAssignments, setSelectedAssignments] = useState([]);
+  const navigate = useNavigate();
+
+  const handleFormulaChange = (event) => {
+    const newFormula = formulas.find(formula => formula.name == event.target.value);
+    setCodeSnippet(newFormula.codeSnippet);
+    setFormula(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    console.log(event.nativeEvent.submitter.name);
+    try {
+      // TODO: send formula to backend -> needs instance id?
+      const formulaObject = ({
+        formula,
+        selectedAssignments
+      });
+      console.log(formulaObject);
+
+      if (event.nativeEvent.submitter.name == 'skipAttributes') {
+        // should this be course id instead? 
+        navigate(`/course-view/${instaceId}`, { replace: true });
+      } else {
+        // TODO: redirect to specify attribures
+      }
+      
+    } catch (exception) {
+      console.log(exception);
+    }
+  };
+
+  const assignmentCheckboxes = () => {
+    return (
+      <>
+        { assignments.map((assignment) => (
+          <FormControlLabel key={assignment.id} control={
+            <Checkbox name={assignment.name} onChange={(event) => {
+              if(event.target.checked) {
+                setSelectedAssignments(prev => [...prev, event.target.name]);
+              } else {
+                setSelectedAssignments(prev => prev.filter(fruit => fruit !== event.target.name));
```

Comment:
fruit? Also maybe since this isn't a one-liner, maybe a separate onChange function above in the component and then just add that in the JSX